pagopa/io-app
# feat: [IOCOM-842] Local message search, new DS

⚠️ This PR depends on #5935 ⚠️

## Short description
This PR implements the new messages local search.

|iOS<br/>Soft keyboard|Android<br/>Hardware keyboard|
|-|-|
|<video src="https://github.com/pagopa/io-app/assets/5150343/732e7561-9b81-4f6e-85b4-20a352625f70"/>|<video src="https://github.com/pagopa/io-app/assets/5150343/a886421f-3cb1-48ef-989c-2eed645b5139"/>|

**_Please note_** that the extra horizontal padding (on list items and dividers) shown in the video above has been removed.

## List of changes proposed in this pull request
- New screen with messages search
- New cached selector for message search

## How to test
Using the io-dev-api-server, generate some messages, make sure to enable the new DS and the new Messages Home and tap the search lens.
